lib/components/FileAttachment/index.jsx

```diff
@@ -0,0 +1,188 @@
+      const uploadedFiles = successful.map(file => ({
+        filename: file.name,
+        signedId: file.response.signed_id,
+        url: file.response.blob_url,
+      }));
+
+      setFileProgresses(prevState => [
+        ...prevState.filter(attachment => attachment.progress !== 100),
+      ]);
+      onChange([...attachments, ...uploadedFiles]);
+      setAdding(false);
+    } catch {}
+  };
+
+  const handleUploadProgress = (file, progress) => {
+    setFileProgresses(prevState =>
+      prevState.map(uploadingFile => ({
+        ...uploadingFile,
+        progress:
+          uploadingFile.filename !== file.name
+            ? uploadingFile.progress
+            : progress.progress,
+      }))
+    );
+  };
+
+  const handleDelete = async ({ signedId }) => {
+    try {
+      await directUploadsApi.destroy(endpoint, signedId);
+      onChange(
+        attachments.filter(attachment => attachment.signedId !== signedId)
+      );
+    } catch {}
+  };
+
+  const handleRename = async (attachment, callback) => {
+    try {
+      const { fileName, signedId } = attachment;
+      const payload = {
+        blob: { filename: fileName },
+      };
+
+      await directUploadsApi.update(endpoint, signedId, payload);
+      onChange(
+        attachments.map(attachment =>
+          attachment.signedId === signedId
+            ? {
+                ...attachment,
+                filename: fileName,
+              }
+            : attachment
+        )
+      );
+      callback(true);
+    } catch {}
+  };
+
+  const handleDownload = ({ url = "", filename = "file.txt" }) => {
+    saveAs(url, filename);
+  };
+
+  const DROPDOWN_OPTIONS = [
+    {
+      label: "Download",
+      handler: handleDownload,
+    },
+    {
+      label: "Rename",
+      handler: handleRename,
+    },
+    {
+      label: "Delete",
+      handler: handleDelete,
+    },
+  ];
+
+  useEffect(() => {
+    if (adding) {
+      uppy.on("upload-progress", handleUploadProgress);
+      handleUpload();
+    }
+
+    return () => {
+      uppy.off("upload-progress", handleUploadProgress);
+    };
+  }, [adding]);
+
+  return (
+    <div className="mt-2">
+      <Typography style="h5">Attachments</Typography>
+      <div className="ne-file-attachments-inner-wrapper">
+        {attachments?.map(attachment => (
+          <Attachment
+            attachment={attachment}
+            dropDownOptions={DROPDOWN_OPTIONS}
+            key={attachment.signedId}
+          />
+        ))}
+        {isUploading &&
+          fileProgresses?.map(attachment => (
+            <AttachmentProgress
+              attachment={attachment}
+              key={attachment.filename}
+            />
+          ))}
+      </div>
+      <div className="upload-btn">
+        <Button
+          label="Add attachments"
+          size="medium"
+          style="link"
+          onClick={() => addFileRef.current.click()}
+        />
+        <input
+          multiple
+          className="file-input"
```

Comment:
Please remove the `className` and `id` from this `input` component. We can directly access it as `input[type=file]` selector from SCSS.